Content.Shared/_CM14/Marines/Skills/SkillsSystem.cs

```diff
@@ -34,6 +44,53 @@ private void OnRequiresSkillBeforeRangedInteract(Entity<RequiresSkillComponent>
         }
     }
 
+    private void OnExamineReagentContainer(Entity<ReagentExaminationRequiresSkillComponent> ent, ref ExaminedEvent args)
+    {
+        if (HasSkills(args.Examiner, in ent.Comp.Skills))
+        {
+            if (!TryComp<SolutionContainerManagerComponent>(args.Examined, out var solutionContainerManager) || solutionContainerManager is null)
+            {
+                return;
+            }
+            var foundReagents = new HashSet<string>();
+            foreach (var solutionContainerID in solutionContainerManager.Containers)
+            {
+                if (!_solutionContainerSystem.TryGetSolution(args.Examined, solutionContainerID, out _, out var solution) || solution is null)
+                {
+                    continue;
+                }
+                foreach (var reagent in solution.Contents)
+                {
+                    var reagentName = _prototypeManager.Index<ReagentPrototype>(reagent.Reagent.Prototype).LocalizedName;
+                    foundReagents.Add(reagentName);
+                }
+            }
+            if (foundReagents.Any())
+            {
+
+                int reagentCount = foundReagents.Count;
+                int i = 0;
+                var fullMessage = "";
+                fullMessage += Loc.GetString("reagents-examine-action-found");
+                fullMessage += " ";
+                foreach (var reagentId in foundReagents)
+                {
+                    fullMessage += reagentId;
+                    if (i > reagentCount)
+                    {
+                        fullMessage += ", ";
+                    }
+                    ++i;
+                }
+                args.PushMarkup(fullMessage);
+            }
+            else
+            {
+                args.PushMarkup(Loc.GetString("reagents-examine-action-found-none"));
+            }
+        }
```

Comment:
The HasSkills and foundReagents.Any checks can be inverted to make the rest of the code less indented, making it easier to read.

Also when using TryComp, we usually put the type next to `out`, which is mostly a code style preference but does matter when using other APIs.
```suggestion
        if (!HasSkills(args.Examiner, in ent.Comp.Skills))
            return;

        if (!TryComp(args.Examined, out SolutionContainerManagerComponent? solutionContainerManager) || solutionContainerManager is null)
        {
            return;
        }
        var foundReagents = new HashSet<string>();
        foreach (var solutionContainerID in solutionContainerManager.Containers)
        {
            if (!_solutionContainerSystem.TryGetSolution(args.Examined, solutionContainerID, out _, out var solution) || solution is null)
            {
                continue;
            }
            foreach (var reagent in solution.Contents)
            {
                var reagentName = _prototypeManager.Index<ReagentPrototype>(reagent.Reagent.Prototype).LocalizedName;
                foundReagents.Add(reagentName);
            }
        }
        if (!foundReagents.Any())
        {
            args.PushMarkup(Loc.GetString("reagents-examine-action-found-none"));
            return;
        }

        int reagentCount = foundReagents.Count;
        int i = 0;
        var fullMessage = "";
        fullMessage += Loc.GetString("reagents-examine-action-found");
        fullMessage += " ";
        foreach (var reagentId in foundReagents)
        {
            fullMessage += reagentId;
            if (i > reagentCount)
            {
                fullMessage += ", ";
            }
            ++i;
        }
        args.PushMarkup(fullMessage);
```